doc/md/fundamentals/2-types/11-results.md

```diff
@@ -0,0 +1,78 @@
+---
+sidebar_position: 11
+---
+
+# Results
+
+| Type    | Syntax                  | Purpose                          | When to use it                                   |
+|---------|-------------------------|----------------------------------|------------------------------------------------|
+| Result  | `Result.Result<T, E>`  where `T` is the success type and `E` is the error type.  | Represents success or failure.   | When an operation can either **succeed or fail**. |
+
+While options are a built-in type, the `Result` is defined as a variant type:
+
+`type Result<Ok, Err> = { #ok : Ok; #err : Err }`
+
+Because of the second type parameter, `Err`, the `Result` type lets you select the type used to describe errors.
+
+Here’s a simple function that validates a username. If it's non-empty, we return a greeting of type `Text`. If not, return an error of type `Text`.
+
+```motoko
+import Result "mo:base/Result";
+
+func greet(name : Text) : Result.Result<Text, Text> {
+  if (name.size() == 0) #err "Name must be present" else #ok ("Hello" # name);
+};
+```
+
+## Pattern matching with `Result`
+
+When a Motoko value has type `Result<T, E>`, it is either a success, written `#ok value`, or a failure, written `#err error`. The fundamental way to access the contents is to use a `switch` expression that explicitly handles both cases.
+
+### `switch`
+
+```motoko
+import Result "mo:base/Result";
+
+func greet(name : Text) : Result.Result<Text, Text> {
+  if (name.size() == 0) #err "Name must be present" else #ok ("Hello " # name);
+};
+
+let result = greet("motoko");
+switch (result){
+  case (#ok message) {
+    debug_show(message) // Hello Motoko
+  };
+  case (#err error){
+    debug_show(error)
+  };
+}
```

Comment:
```suggestion
let result = greet("motoko");
switch (result){
  case (#ok message) {
    message
  };
  case (#err error){
    error
  };
} // "Hello Motoko"
```
The debug_show are redundant since message and error already have type Text.